ark/phenotyping/pixel_cluster_utils.py

```diff
@@ -317,10 +327,29 @@ def compute_pixel_cluster_channel_avg(fovs, channels, base_dir, pixel_cluster_co
         valid_cluster_cols=['pixel_som_cluster', 'pixel_meta_cluster']
     )
 
-    # define the cluster averages DataFrame
-    cluster_avgs = pd.DataFrame()
+    # verify num_pixel_clusters is valid
+    if num_pixel_clusters <= 0:
+        raise ValueError("Number of pixel clusters desired must be a positive integer")
 
-    for fov in fovs:
+    # verify fovs subset value is valid
+    if num_fovs_subset <= 0:
+        raise ValueError("Number of fovs to subset must be a positive integer")
+
+    # define a list to hold the cluster averages for each FOV
+    fov_cluster_avgs = []
+
+    # warn the user that we can only select so many FOVs if len(fovs) < num_fovs_subset
+    if len(fovs) < num_fovs_subset:
+        warnings.warn(
+            'Provided num_fovs_subset=%d but only %d FOVs in dataset, '
+            'subsetting just the %d FOVs' %
+            (num_fovs_subset, len(fovs), len(fovs))
+        )
+
+    # subset number of FOVs based on num_fovs_subset if less than total number of fovs
+    fovs_sub = random.sample(fovs, num_fovs_subset) if num_fovs_subset < len(fovs) else fovs
```

Comment:
Do we want to add a seed here to make it reproducible?

Reply:
Yup, that's a good idea.